### PR TITLE
Milestone picker: show all milestones in a scrollable panel #1413

### LINKED ISSUE #1413: Milestone picker: show all milestones in a scrollable panel
Below the best 5 matches panel, we can show the full list in a scrollable panel. It can be a static panel (no highlighting etc), but it should be clickable.

### PR DESCRIPTION
**Pull request by garbanzos**
Fixes #1413. 

image

### CODE REVIEW

**Comment by garbanzos** [comment]
@ndt93 Could you assign a reviewer for this?

**Comment by ndt93** [comment]
@dariusf Hey darius, can help review this? Thanks.

**Comment by garbanzos** [comment]
@dariusf ready for review.

File: src/main/java/ui/components/pickers/MilestonePickerDialog.java [hunk 1]
```diff
@@ -126,14 +128,30 @@ private void initUI() {
         assignedMilestoneBox = createAssignedMilestoneBox();
         matchingMilestonesBox = createMatchingMilestonesBox();
         inputField = createInputField();
+        ScrollPane allMilestoneScrollPane = createAllMilestonesScrollPane();
 
-        milestoneDialogBox.getChildren().addAll(assignedMilestoneBox, inputField, matchingMilestonesBox);
+        milestoneDialogBox.setSpacing(5);
```
Comment (dariusf) [inline src/main/java/ui/components/pickers/MilestonePickerDialog.java]: Maybe this bit should be inside `createMilestoneDialogBox`?

File: src/main/java/ui/components/pickers/MilestonePickerDialog.java [hunk 2]
```diff
@@ -126,14 +128,30 @@ private void initUI() {
...
+        milestoneDialogBox.getChildren().addAll(assignedMilestoneBox, inputField,
+                new Label(TITLE_RESULT), matchingMilestonesBox, new Label(TITLE_ALL), allMilestoneScrollPane);
```
Comment (dariusf) [inline src/main/java/ui/components/pickers/MilestonePickerDialog.java]: Indentation can be improved a bit here, to show that all these nodes are siblings.

File: src/main/java/ui/components/pickers/MilestonePickerDialog.java [hunk 3]
```diff
@@ -126,14 +128,30 @@ private void initUI() {
...
         getDialogPane().setContent(milestoneDialogBox);
         refreshUI(state);
         Platform.runLater(inputField::requestFocus);
     }
 
+    private ScrollPane createAllMilestonesScrollPane() {
+        VBox milestonesBox = new VBox();
+        milestonesBox.setStyle("-fx-border-radius: 3;-fx-background-color: white;-fx-border-color: black;");
+        originalMilestones.forEach( milestone -> milestonesBox.getChildren().add(
+                setMouseClickForNode(milestone.getDetailedMilestoneNode(), milestone.getTitle()))
```
Comment (dariusf) [inline src/main/java/ui/components/pickers/MilestonePickerDialog.java]: There's extra whitespace before the lambda. Also, maybe indent like this?
```java
milestones.forEach(milestone ->
    milestonesBox.getChildren().add(
        // ...
```

**Comment by dariusf** [comment]
Mostly just cosmetic changes.

**Comment by garbanzos** [comment]
@dariusf Ready for review again!

**Comment by damithc** [comment]
Can the user pick a milestone by clicking an item in the scrollable panel? If not, need to make it so.

**Comment by garbanzos** [comment]
@Yeap the milestones in the scrollpane is clickable

**Comment by damithc** [comment]
Noted. Go ahead and merge.

**Pull request merged**